src/commands/upgrade.ts

```diff
@@ -75,6 +75,9 @@ export class UpgradeCommand extends Command {
         }
 
         const upgradePrice = condition.upgradePrice * (card.foil ? 2 : 1);
+        const pityFactor = 0.07 * card.pity;
```

Comment:
This should probably not be linear, rather something that slowly approaches but never reaches 1

Reply:
Can agree with making it non-linear, though making it "slowly" approach anything seems a bit pointless since most upgrade sequences currently don't even reach 10 total rolls
; this isn't like spending 100 gacha pulls to get your guaranteed SSR character

Reply:
How about this? It's `pity / (pity + 8)`

<img width="576" alt="image" src="https://github.com/user-attachments/assets/c14b4548-c578-4c4a-bb0c-387e3b7270da">

Reply:
Comparison with previous formula

<img width="542" alt="image" src="https://github.com/user-attachments/assets/d877a1dd-1437-4c5c-929d-d223b62e48f7">

Reply:
simulated this and got this, it going up steeper early has a stronger effect it seems
but having the average coincidentally hover around the price of a superdrop seems rather neat

| | Costs |
|----------|------|
| Mean     | 1501 |
| Median   | 1300 |
| Std Dev  | 713  |